File: 0049-group-anagrams/0049-group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
 const map = new Map();

  for (let word of strs) {
    const sorted = word.split('').sort().join(''); // key
    if (!map.has(sorted)) {
      map.set(sorted, []);
    }
    map.get(sorted).push(word);
  }

  return Array.from(map.values());
    
};

File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
         unordered_map<string,vector<string>> anagramGroups;

        for(string word : strs){
        
           string sortedWord = word;
           sort(sortedWord.begin(),sortedWord.end());
           anagramGroups[sortedWord].push_back(word);
        }

        // convert map to array

        vector<vector<string>> result;

        for(auto& pair : anagramGroups){
            result.push_back(pair.second);
        }

      return result;
    }
};